created new FAQ from various FAQ files in sigstore docs and website. (#183)
sigstore/docs: content/en/faq.md

This FAQ is intended to go as in depth as possible for anyone using sigstore. 

## General

> ### "What security checks do you use internally?

Reviewer: ```suggestion
### What security checks do you use internally?
```

### What is the relationship between Sigstore and TUF?

Sigstore's root of trust for Fulcio and Rekor are distributed using The Update Framework [TUF](https://theupdateframework.io).  The releases for Cosign, Rekor, and Fulcio are also signed by a key distributed with TUF.  The projects solve a similar problem of signing and verifying software that will be distributed via a repository. Sigstore provides key signing using multiple techniques, including without requiring developers to manage keys by using OIDC.  In contrast, TUF typically requires keys to be managed by developers, but addresses a threat model where all online servers and keys could be compromised.  The projects are collaborating and can be used cooperatively.  More guidance on this will be forthcoming as the communities collaborate on their first large co-deployments.

## Cosign

> ### Why not use Notary v2?

Reviewer: Maybe this can just be about what are the differences between Notary and Cosign vs. why not use an alternate tool?

Author: This is actually a long discussion.  So I added a quick sentence about differences and pointed to the page that discusses the issue in-depth.

> The differences right now between Notary V2 and Cosignare in three main parts — Signature Formats, Signature Discovery, and Key Management. For a complete discussion, see [Notary V2 and Cosign](https://medium.com/@dlorenc/notary-v2-and-cosign-b816658f044d)

Reviewer: ```suggestion
The differences right now between Notary V2 and Cosign are in three main parts — Signature Formats, Signature Discovery, and Key Management. For a complete discussion, review [Notary V2 and Cosign](https://medium.com/@dlorenc/notary-v2-and-cosign-b816658f044d).
```

### Why not use containers/image signing?

`containers/image` signing is close to `cosign`, and we reuse payload formats.
`cosign` differs in that it signs with ECDSA-P256 keys instead of PGP, and stores
signatures in the registry.

### Why not use $FOO?

We designed `cosign` to meet a few specific requirements.  The design requirements are:

* No external services for signature storage, querying, or retrieval
* We aim for as much registry support as possible
* Everything should work over the registry API

> * PGP should not be required at all.

Reviewer: Remove full stop for consistency

```suggestion
* PGP should not be required at all
```

containing a SHA256 hash of the commit SHA, as well as the code signing
certificate. Review guidance on
[Verifying the Transparency Log](/rekor/public-instance/#auditing-the-public-instance) for more
details.

By default, data is written to the
[public Rekor instance](/rekor/public-instance/). In
particular, users and organizations may be sensitive to the data contained
within code signing certificates returned by Fulcio, which may include user
emails or repository identifiers. Review
[OIDC Usage in Fulcio](/fulcio/oidc-in-fulcio/) for more details regarding what
data is contained in the code signing certs. Alternately, you can learn how to
[Deploy a Rekor Server Manually](/rekor/installation/#deploy-a-rekor-server-manually),
which would set up your own Rekor instance.

### Why does a browser window open for each commit in a rebase?

For Git, each commit in a rebase is considered a distinct signing operation so
by default an ephemeral key is generated for each commit. There are a
few options to help automating the authentication process:

- Setting the [`connectorID`](/gitsign/usage/#configuration) value can be set to

>   automatically select the desired provider for Dex backed OIDC providers

Reviewer: ```suggestion
  automatically select the desired provider for Dex-backed OIDC providers
```

  (including the public Sigstore instance at `oauth.sigstore.dev`). While this
  still requires a browser window to open, this does not require an extra click
  to select the provider.
- Starting in v0.2.0, Gitsign has experimental support for key caching to allow
  users to reuse ephemeral keys for the lifetime of the Fulcio certificate. If

>   you are interested, check out the

Reviewer: ```suggestion
  you are interested in learning more, check out the
```

  [`gitsign-credential-cache` README](https://github.com/sigstore/gitsign/tree/main/cmd/gitsign-credential-cache).

## Rekor

### Is the transparency log monitored?

> Purdue University operates a log monitor, checking that the log remains append-only. Sigstore provides an easy-to-use GitHub Actions-based monitor for consistency and identity at https://github.com/sigstore/rekor-monitor.

Reviewer: ```suggestion
Purdue University operates a log monitor, checking that the log remains append-only. Sigstore provides an easy-to-use GitHub Actions-based monitor for consistency and identity at [https://github.com/sigstore/rekor-monitor](https://github.com/sigstore/rekor-monitor).
```

### Can I run my own transparency log monitor?

> We would love that. The more people monitoring the logs and providing useful services to users, the better. Check out https://github.com/sigstore/rekor-monitor for a GitHub Actions based monitor.

Reviewer: ```suggestion
We would love that. The more people monitoring the logs and providing useful services to users, the better. Check out [https://github.com/sigstore/rekor-monitor](https://github.com/sigstore/rekor-monitor) for a GitHub Actions based monitor.
```

  
### Why is Rekor centralized?

> There's no need for a distributed source of transparency as there can be multiple points of transparency which only adds more sources of security guarantee, not less. An entity can post to as many Rekors as they want and inform users of where they post. We do encourage folks to use common public instances, but we don't seek to enforce this. We do plan to look to produce a gossip protocol, for those that desire a more decentralised model (if there's demand).

Reviewer: ```suggestion
There's no need for a distributed source of transparency as there can be multiple points of transparency which only adds more sources of security guarantee, not fewer. An entity can post to as many Rekor logs as they want and inform users of where they post. We do encourage folks to use common public instances, but we don't seek to enforce this. We do plan to look to produce a gossip protocol, for those that desire a more decentralised model (if there's demand).
```

This FAQ is intended to go as in depth as possible for anyone using sigstore. 

## General

### "What security checks do you use internally?
 
We’ve adopted a security disclosures and response policy to make sure we can responsibly handle critical issues. We have an initial Security Response Committee, who for each vulnerability reported will coordinate to create the fix and release, and keep the committee looped in. The policy in full is here: <https://github.com/sigstore/.github/blob/main/SECURITY.md>.

### How does Sigstore integrate in-toto?

Sigstore is co-led by Santiago Torres-Arias, the lead of the in-toto project, and the two efforts are complementary. Sigstore supports and uses in-toto in a number of ways. Cosign can be used to generate and verify in-toto attestations and Rekor can store those attestations. in-toto also supports the use of PKI semantics for signing metadata which will be used in future to support Fulcio for developer key management.

### What is the relationship between Sigstore and TUF?

> Sigstore's root of trust for Fulcio and Rekor are distributed using TUF.  The releases for Cosign, Rekor, and Fulcio are also signed by a key distributed with TUF.  The projects solve a similar problem of signing and verifying software that will be distributed via a repository. Sigstore provides key signing using multiple techniques, including without requiring developers to manage keys by using OIDC.  In contrast, TUF typically requires keys to be managed by developers, but addresses a threat model where all online servers and keys could be compromised.  The projects are collaborating and can be used cooperatively.  More guidance on this will be forthcoming as the communities collaborate on their first large co-deployments.

Author: done

## Cosign

> ### Can I use Cosign to sign things *besides* OCI container images?

Author: removed

Yes! See [Signing with Blobs.](../signing_with_blobs) But Cosign *does* have great support for OCI registries.

> ### Why not use Notary v2

Reviewer 2: ```suggestion
### Why not use Notary v2?
```

> ### Why not use Notary v2

Author: added a bit more to the response and reworded how the link works, and removed the part about filing a PR.

It's hard to answer this briefly. This post contains some comparisons:

[Notary V2 and Cosign](https://medium.com/@dlorenc/notary-v2-and-cosign-b816658f044d)

If you find other comparison posts, please send a PR here and we'll link them all.

> ### Why not use containers/image signing

Reviewer 2: ```suggestion
### Why not use containers/image signing?
```

`containers/image` signing is close to `cosign`, and we reuse payload formats.
`cosign` differs in that it signs with ECDSA-P256 keys instead of PGP, and stores
signatures in the registry.

### Why not use $FOO?

> See [Requirements](#design-requirements).

Author: took out the header and the link.   The requirements now just directly follow the previous section.

We designed `cosign` to meet a few specific requirements, and didn't find anything else that met all of these.
If you're aware of another system that does meet these, please let us know!

> ### What are the design requirements?

Author: Reworded for clairty.

Gitsign records commits and the code signing certificates to
[Rekor](/rekor/overview/). This data is a
[HashedRekord](https://github.com/sigstore/rekor/blob/e375eb461cae524270889b57a249ff086bea6c05/types.md#hashed-rekord)
containing a SHA256 hash of the commit SHA, as well as the code signing
certificate. Review guidance on
[Verifying the Transparency Log](/rekor/public-instance/#auditing-the-public-instance) for more
details.

By default, data is written to the
[public Rekor instance](/rekor/public-instance/). In
particular, users and organizations may be sensitive to the data contained
within code signing certificates returned by Fulcio, which may include user
emails or repository identifiers. Review
[OIDC Usage in Fulcio](/fulcio/oidc-in-fulcio/) for more details regarding what
data is contained in the code signing certs. Alternately, you can learn how to
[Deploy a Rekor Server Manually](/rekor/installation/#deploy-a-rekor-server-manually),
which would set up your own Rekor instance.

### Why does a browser window open for each commit in a rebase?

For Git, each commit in a rebase is considered a distinct signing operation so

> by default an ephemeral key is generated for each commit. To help, there are a

Reviewer 2: ```suggestion
by default an ephemeral key is generated for each commit. There are a
```

few options to help automating the authentication process:

- Setting the [`connectorID`](/gitsign/usage/#configuration) value can be set to
  automatically select the desired provider for Dex backed OIDC providers
  (including the public Sigstore instance at `oauth.sigstore.dev`). While this
  still requires a browser window to open, this does not require an extra click
  to select the provider.
- Starting in v0.2.0, Gitsign has experimental support for key caching to allow
  users to reuse ephemeral keys for the lifetime of the Fulcio certificate. If
  you are interested, check out the
  [`gitsign-credential-cache` README](https://github.com/sigstore/gitsign/tree/main/cmd/gitsign-credential-cache).

## Rekor

### Is the transparency log monitored?

Purdue University operates a log monitor, checking that the log remains append-only. Sigstore provides an easy-to-use GitHub Actions-based monitor for consistency and identity at https://github.com/sigstore/rekor-monitor.

### Can I run my own transparency log monitor?

> We would love that, you’re definitely encouraged to do so. The more people monitoring the logs and providing useful services to users, the better. Check out https://github.com/sigstore/rekor-monitor for a GitHub Actions based monitor.

Reviewer 2: ```suggestion
We would love that. The more people monitoring the logs and providing useful services to users, the better. Check out https://github.com/sigstore/rekor-monitor for a GitHub Actions based monitor.
```

  
### Why is Rekor centralized?

> - Well, why not? There's no need for a distributed source of transparency. There can be multiple points of transparency which only adds more sources of security guarantee, not less. An entity can post to as many Rekors as they want and inform users of where they post. We do encourage folks to use common public instances, but we don't seek to enforce this. We do plan to look to produce a gossip protocol, for those that desire a more decentralised model (if the demand is shown).

Reviewer 2: ```suggestion
- Well, why not? There's no need for a distributed source of transparency. There can be multiple points of transparency which only adds more sources of security guarantee, not less. An entity can post to as many Rekors as they want and inform users of where they post. We do encourage folks to use common public instances, but we don't seek to enforce this. We do plan to look to produce a gossip protocol, for those that desire a more decentralised model (if there's demand).
```

> - Well, why not? There's no need for a distributed source of transparency. There can be multiple points of transparency which only adds more sources of security guarantee, not less. An entity can post to as many Rekors as they want and inform users of where they post. We do encourage folks to use common public instances, but we don't seek to enforce this. We do plan to look to produce a gossip protocol, for those that desire a more decentralised model (if the demand is shown).

Author: Removed well why not.
